### PR TITLE
closed #3630:Uses a struct of TTF configuration for Label::createWithTTF  to reduce parameters and make this interface more easily to use.

### PR DESCRIPTION
**Pull request by WenhaiLin**
1. Uses a struct for TTF configuration for Label::createWithTTF to reduce parameters
2. Add create/setTTFConfig/setBMFontFilePath for make this interface more easily to use.

### CODE REVIEW

File: cocos/2d/CCLabel.h [hunk 1]
```diff
@@ -54,23 +54,45 @@ enum class LabelEffect {
...
 class FontAtlas;
 
-
+typedef struct _ttfConfig
+{
+    std::string fontFilePath;
+    int fontSize;
+    GlyphCollection glyphs;
+    const char *customGlyphs;
+    bool distanceFieldEnable;
```
Comment (minggo) [inline cocos/2d/CCLabel.h]: distanceFieldEnable -> distanceFieldEnabled

File: cocos/2d/CCLabel.h [hunk 2]
```diff
@@ -54,23 +54,45 @@ enum class LabelEffect {
 struct FontLetterDefinition;
 class FontAtlas;
 
-
+typedef struct _ttfConfig
+{
+    std::string fontFilePath;
+    int fontSize;
+    GlyphCollection glyphs;
+    const char *customGlyphs;
```
Comment (minggo) [inline cocos/2d/CCLabel.h:62]: Why use const char*?

File: cocos/2d/CCLabel.cpp [hunk 3]
```diff
@@ -118,6 +109,7 @@ Label* Label::createWithAtlas(FontAtlas *atlas, TextHAlignment alignment, int li
 , _fontSize(0)
 , _uniformEffectColor(0)
 {
+    _cascadeColorEnabled = true;
```
Comment (dumganhar) [inline cocos/2d/CCLabel.cpp:111]: Why not to add it after the line `, _uniformEffectColor(0)`?

Comment (WenhaiLin) [inline cocos/2d/CCLabel.cpp:111]: _cascadeColorEnabled is not label's member variable

File: cocos/2d/CCLabel.h (hunk 2 above)
Comment (WenhaiLin) [inline cocos/2d/CCLabel.h:62]: It mean create  two string when the formal parameter is std::string& and the  actual parameter is char*.

Comment (WenhaiLin) [inline cocos/2d/CCLabel.h:62]: In most cases,the actual parameter is char*.

**Comment by dumganhar** [comment]
Thanks.

**Pull request merged**